app/services/objective_notification_service.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class ObjectiveNotificationService
+  def perform
+    Objective.visible.each do |objective|
+      automatic_topup_amount = objective.account.automatic_topup_configs.sum(:amount)
+      account_balance = objective.account.balance
+      objective_amount = objective.amount
+      total_weeks_left = ([objective_amount - account_balance, 0].max / automatic_topup_amount).to_i
```

Comment:
Can we reuse the calculations used in the objectives table? I mean is there a DRY solution?